composer.json

```diff
@@ -15,7 +15,10 @@
         "psr-4": {"SwotPHP\\": "src/"}
     },
     "require": {
-        "php": ">=5.3.0",
-        "jeremykendall/php-domain-parser": "~1.3"
+        "php": ">=7.2",
+        "jeremykendall/php-domain-parser": "~1.4"
```

Comment:
I think the underlying issues causing the failures in php 7.4 are coming from within this library. Unfortunately it looks like v5.0 of this library is very different than ~1.3.